### PR TITLE
Remove "latest" from GCP reference and update installation instructions

### LINKED ISSUE #710: Remove "latest" from GCP and update documentation
For historical reasons, we have a "latest" directory on GCP. We don't currently have automation to publish to that directory, so it doesn't stay up-to-date.

Now that we're reliably publishing GitHub releases, we should update documentation to refer to those instead, and remove the "latest" directory.

### PR DESCRIPTION
**Pull request by navidshaikh**
**What this PR does / why we need it**:
- Remove `latest` from GCP reference
- Add instructions to install cli and plugins from release page

Signed-off-by: Navid Shaikh <navids@vmware.com>

**Which issue(s) this PR fixes**:

Fixes #710 

**Describe testing done for PR**:

- Tested the steps on MacOS
- Tested the steps on Windows

**Special notes for your reviewer**:

**Release note**:

```release-note

```
**New PR Checklist**

- [x] Ensure PR contains only public links or terms
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Squash the commits in this branch before merge to preserve our git history
- [x] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [x] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

File: docs/cli/getting-started.md [hunk 1]
```diff
@@ -9,59 +9,114 @@ A simple set of instructions to set up and use the Tanzu CLI.
...
-    chmod +x /usr/local/bin/tanzu
-```
+  ```sh
+  mkdir tanzu && tar -xvf tanzu-framework-darwin-amd64.tar -C tanzu
+  ```
+
+- Install the `tanzu` CLI
+
+  ```sh
+  sudo install tanzu/cli/core/v0.5.0/tanzu-core-darwin_amd64 /usr/local/bin/tanzu
```
Comment (vuil) [inline docs/cli/getting-started.md]: this references mac tarball in the mac/linux section

File: docs/cli/getting-started.md [hunk 2]
```diff
@@ -9,59 +9,114 @@ A simple set of instructions to set up and use the Tanzu CLI.
...
 
-```bash
-./artifacts/darwin/amd64/cli/core/latest/tanzu-core-darwin_amd64
-```
+  ```bash
+  ./artifacts/darwin/amd64/cli/core/latest/tanzu-core-darwin_amd64
+  ```
 
-If you additionaly want to build and install CLI and plugins for all platforms, run:
+- If you additionaly want to build and install CLI and plugins for all platforms, run:
```
Comment (vuil) [inline docs/cli/getting-started.md:131]:
```suggestion
- If you additionally want to build and install CLI and plugins for all platforms, run:
```

**Review by vuil — APPROVED** [review]
lgtm, just a small nit, and found a preexisting typo.

File: docs/cli/getting-started.md (hunk 1 above)
Comment (navidshaikh) [reply docs/cli/getting-started.md]: updated

**Review by rajaskakodkar — APPROVED** [review]
Validated these changes locally, could successfully install tanzu cli. 

lgtm

**Pull request merged**